examples/python/Boxes_QPE_example.py

```diff
@@ -0,0 +1,561 @@
+    state_prep_register = qpe_circ.add_q_register("p", n_ancillas)
+    c_reg = qpe_circ.add_c_register("c", n_measurement_qubits)
+    
+    state_prep_qubits = list(state_prep_register)
+    measurement_qubits = list(measurement_register)
+    
+    qpe_circ.add_circuit(state_prep_circuit, state_prep_qubits)
+    
+    unitary_circuit.name = "U"
+    u_box = CircBox(unitary_circuit)
+    controlled_u = QControlBox(u_box, 1)
+    
+    for m_qubit in measurement_qubits:
+        qpe_circ.H(m_qubit)
+        
+    for m_qubit in range(n_measurement_qubits):
+        control_index = n_measurement_qubits - m_qubit - 1
+        control_qubit = [measurement_register[control_index]]
+        for _ in range(2**m_qubit):
+                qpe_circ.add_qcontrolbox(
+                    controlled_u, control_qubit + state_prep_qubits
+                )
+        
+    qft_box = CircBox(build_qft_circuit(n_measurement_qubits))
+    inverse_qft_box = qft_box.dagger
+    
+    qpe_circ.add_circbox(inverse_qft_box, measurement_qubits)
+    
+    qpe_circ.measure_register(measurement_register, "c")
+    
+    return qpe_circ
+
+
+# ## Phase Estimation with a Trivial Eigenstate
+# 
+# Lets test our circuit construction by preparing a trivial $|1\rangle $eigenstate of the $\text{U1}$ gate. We can then see if our phase estimation circuit returns the expected eigenvalue.
+
+# $$
+# \begin{equation}
+# U1(\phi)|1\rangle = e^{i\phi} = e^{2 \pi i \theta} \implies \theta = \frac{\phi}{2}
+# \end{equation}
+# $$
+# 
+# So we expect that our ideal phase $\theta$ will be half the input angle $\phi$ to our $U1$ gate.
+
+# In[15]:
+
+
+state_prep_circuit = Circuit(1).X(0)
+
+input_angle = 0.73
+
+unitary_circuit = Circuit(1).add_gate(OpType.U1, [input_angle], [0])
+
+
+# In[16]:
+
+
+qpe_circ_trivial = build_phase_est_circuit(4, state_prep_circuit=state_prep_circuit, unitary_circuit=unitary_circuit)
+
+
+# In[17]:
+
+
+render_circuit_jupyter(qpe_circ_trivial)
+
+
+# Lets use the noiseless `AerBackend` simulator to run our phase estimation circuit.
+
+# In[18]:
+
+
+from pytket.extensions.qiskit import AerBackend
+
+backend = AerBackend()
+
+compiled_circ = backend.get_compiled_circuit(qpe_circ_trivial)
+
+
+n_shots = 1000
+result = backend.run_circuit(compiled_circ, n_shots)
+
+print(result.get_counts())
+
+
+# In[19]:
+
+
+from plotting import plot_qpe_results
+
+plot_qpe_results(result, y_limit=1.2*n_shots)
+
+
+# As expected we see one outcome with high probability. Lets now extract our approximation of $\theta$ from our output bitstrings.
+# 
+# suppose the $j$ is an integer representation of our most commonly measured bitstring.
+
+# $$
+# \begin{equation}
+# \theta_{estimate} = \frac{j}{N}
+# \end{equation}
+# $$
+
+# Here $N = 2 ^n$ where $n$ is the number of measurement qubits.
+
+# In[20]:
+
+
+from pytket.backends.backendresult import BackendResult
+
+def single_phase_from_backendresult(result: BackendResult) -> float:
```

Comment:
I'd condense slightly (also you return an int but the type signature says float)
```def single_phase_from_backendresult(result: BackendResult) -> int:
    
    # Extract most common measurement outcome
    basis_state = result.get_counts().most_common()[0][0] 
    bitstring = "".join([str(bit) for bit in basis_state])
    
    # Calculate theta
    return int(bitstring, 2) / (2**len(bitstring))```

Reply:
Only the numerator here is coerced to an int. dividing an int by another int will give you the approximation of the phase

Reply:
Also I've done the recommended simplification